third_party/move/move-binary-format/src/check_bounds.rs

```diff
@@ -244,6 +245,66 @@ impl<'a> BoundsChecker<'a> {
         let type_param_count = function_handle.type_parameters.len();
         self.check_type_parameters_in_signature(function_handle.parameters, type_param_count)?;
         self.check_type_parameters_in_signature(function_handle.return_, type_param_count)?;
+        // access specifiers must be in bounds
+        if let Some(specs) = &function_handle.access_specifiers {
+            for spec in specs {
+                self.check_access_specifier(function_handle, spec)?
+            }
+        }
+        Ok(())
+    }
+
+    fn check_access_specifier(
+        &self,
+        function: &FunctionHandle,
+        spec: &AccessSpecifier,
+    ) -> PartialVMResult<()> {
+        let AccessSpecifier {
+            kind: _,
+            negated: _,
+            resource,
+            address,
+        } = spec;
+        match resource {
+            ResourceSpecifier::Any => {},
+            ResourceSpecifier::DeclaredAtAddress(addr_idx) => {
+                check_bounds_impl(self.view.address_identifiers(), *addr_idx)?
+            },
+            ResourceSpecifier::DeclaredInModule(mod_idx) => {
+                check_bounds_impl(self.view.module_handles(), *mod_idx)?
+            },
+            ResourceSpecifier::Resource(stu_idx) => {
+                check_bounds_impl(self.view.struct_handles(), *stu_idx)?
+            },
+            ResourceSpecifier::ResourceInstantiation(stu_idx, sig_idx) => {
+                check_bounds_impl(self.view.struct_handles(), *stu_idx)?;
+                check_bounds_impl(self.view.signatures(), *sig_idx)?;
+            },
+        }
+        match address {
+            AddressSpecifier::Any => {},
+            AddressSpecifier::Literal(addr_idx) => {
+                check_bounds_impl(self.view.address_identifiers(), *addr_idx)?
+            },
+            AddressSpecifier::Parameter(loc_idx, fun_opt) => {
+                let sign = self.view.signature_at(function.parameters);
```

Comment:
Based on documentation:
```
/// The index of a parameter of the current function. If `modifier` is not given, the
/// parameter must have address type. Otherwise `modifier` must be a function which takes
/// a value (or reference) of the parameter type and delivers an address.
```
I am not familiar with this code, where do we check that the type at index in `sign` is address iff `fun_opt.is_none()`? Where do we check the function signature that it returns an address?